specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-10-03-preview/HybridCompute.json

```diff
@@ -0,0 +1,4673 @@
+      "description": "Describes the properties of a License."
+    },
+    "LicenseType": {
+      "type": "string",
+      "enum": [
+        "ESU"
+      ],
+      "x-ms-enum": {
+        "name": "LicenseType",
+        "modelAsString": true
+      },
+      "description": "The type of the license resource."
+    },
+    "LicenseState": {
+      "type": "string",
+      "enum": [
+        "Activated",
+        "Deactivated"
+      ],
+      "x-ms-enum": {
+        "name": "LicenseState",
+        "modelAsString": true
+      },
+      "description": "Describes the state of the license."
+    },
+    "LicenseEdition": {
+      "type": "string",
+      "enum": [
+        "Standard",
+        "Datacenter"
+      ],
+      "x-ms-enum": {
+        "name": "LicenseEdition",
+        "modelAsString": true
+      },
+      "description": "Describes the edition of the license. The values are either Standard or Datacenter."
+    },
+    "LicenseTarget": {
+      "type": "string",
+      "enum": [
+        "Windows Server 2012",
+        "Windows Server 2012 R2"
+      ],
+      "x-ms-enum": {
+        "name": "LicenseTarget",
+        "modelAsString": true
+      },
+      "description": "Describes the license target server."
+    },
+    "LicenseCoreType": {
+      "type": "string",
+      "enum": [
+        "pCore",
+        "vCore"
+      ],
+      "x-ms-enum": {
+        "name": "LicenseCoreType",
+        "modelAsString": true
+      },
+      "description": "Describes the license core type (pCore or vCore)."
+    },
+    "LicensesListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/License"
+          },
+          "description": "The list of licenses."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of Machines. Call ListNext() with this URI to fetch the next page of license profile."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List license operation response."
+    },
+    "LicenseProfile": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "properties": {
+            "softwareAssurance": {
+              "x-ms-client-flatten": true,
+              "type": "object",
+              "properties": {
+                "softwareAssuranceCustomer": {
+                  "type": "boolean",
+                  "description": "Specifies if this machine is licensed as part of a Software Assurance agreement."
+                }
+              }
+            },
+            "esuProfile": {
+              "x-ms-client-flatten": true,
+              "$ref": "#/definitions/LicenseProfileArmEsuProperties",
+              "description": "Hybrid Compute ESU Profile properties"
+            },
+            "osProfile": {
```

Comment:
looking this over again, we should probably find a different name for this, as the machine resource already has an "osProfile" property, so if the licenseProfile has another "osProfile", that could be confusing.

Reply:
We'll change it to **ProductProfile**. Also change the **osType** to **productType**. Sounds better?

Reply:
works for me

Reply:
Updated.